guide.md

```diff
@@ -0,0 +1,388 @@
+Getting Started 2.0
+===
+
+For this guide, we're going to walk through some simple examples for each of the
+pieces of CRUD. Each step in this guide will build on the previous, and is meant
+to be followed along. This guide assumes that you're using PostgreSQL on
+nightly. We'll talk about how to use Diesel on stable in the final chapter.
+Before we start, make sure you have PostgreSQL installed and running.
+
+The first thing we need to do is generate our project.
+
+```shell
+cargo new diesel_demo
+cd diesel_demo
+```
+
+We're going to use a [`.env`][dotenv-rust] to manage our environment variables
+for us, and [`diesel_cli`][diesel-cli] to help set up the database.
+
+[dotenv-rust]: https://github.com/slapresta/rust-dotenv
+[diesel-cli]: https://github.com/sgrif/diesel/diesel_cli
+
+```shell
+echo DATABASE_URL=postgres://localhost/diesel_demo > .env
+cargo install diesel_cli
+diesel setup
+```
+
+This will create our database for us (if it didn't already exist), and create an
+empty migrations directory that we can use to manage our schema. More on that
+later. Let's add Diesel to our `Cargo.toml`. We'll also add `dotenv` so we can
```

Comment:
The ordering of this all took me a bit by surprise because we did something related to `dotenv` already (albeit, we just added it's file) and only now are we declaring the dependency.

I gather that's possibly because `diesel_cli` depends on .env internally?